hazelcast/src/test/java/com/hazelcast/test/mocknetwork/MockJoiner.java

```diff
@@ -50,7 +50,8 @@ public void doJoin() {
 
         Address previousJoinAddress = null;
         long joinAddressTimeout = 0;
-        while (shouldRetry() && (Clock.currentTimeMillis() - joinStartTime < maxJoinMillis)) {
+        boolean shouldRetry;
```

Comment:
Similar thing here:
```java
while (Clock.currentTimeMillis() - joinStartTime < maxJoinMillis) {
    synchronized (registry) {
        Address joinAddress = getJoinAddress();
        verifyInvariant(joinAddress != null, "joinAddress should not be null");

        if (!joinAddress.equals(previousJoinAddress)) {
            previousJoinAddress = joinAddress;
            joinAddressTimeout = Clock.currentTimeMillis() + JOIN_ADDRESS_TIMEOUT_IN_MILLIS;
        }

        if (node.getThisAddress().equals(joinAddress)) {
            logger.fine("This node is found as master, no need to join.");
            clusterJoinManager.setThisMemberAsMaster();
            break;
        }

        logger.fine("Sending join request to " + joinAddress);
        if (!clusterJoinManager.sendJoinRequest(joinAddress, true)) {
            logger.fine("Could not send join request to " + joinAddress);
            clusterService.setMasterAddressToJoin(null);
        }

        if (Clock.currentTimeMillis() > joinAddressTimeout) {
            logger.warning("Resetting master address because join address timeout");
            previousJoinAddress = null;
            joinAddressTimeout = 0;
            clusterService.setMasterAddressToJoin(null);
        }
    }
    if (!shouldRetry()) {
        break;
    }
    try {
        Thread.sleep(100);
    } catch (InterruptedException e) {
        e.printStackTrace();
        break;
    }
}
```